src/sql/workbench/contrib/notebook/browser/notebookExplorer/notebookExplorerViewlet.ts

```diff
@@ -90,30 +125,273 @@ export class NotebookExplorerViewPaneContainer extends ViewPaneContainer {
 		@IConfigurationService configurationService: IConfigurationService,
 		@IMenuService private menuService: IMenuService,
 		@IContextKeyService private contextKeyService: IContextKeyService,
-		@IViewDescriptorService viewDescriptorService: IViewDescriptorService
+		@IViewDescriptorService viewDescriptorService: IViewDescriptorService,
+		@IFileService private readonly fileService: IFileService,
 	) {
 		super(VIEWLET_ID, { mergeViewWithContainerWhenSingleView: true }, instantiationService, configurationService, layoutService, contextMenuService, telemetryService, extensionService, themeService, storageService, contextService, viewDescriptorService);
+		this.inputBoxFocused = Constants.InputBoxFocusedKey.bindTo(this.contextKeyService);
+		this.triggerQueryDelayer = this._register(new Delayer<void>(0));
+		this.queryBuilder = this.instantiationService.createInstance(QueryBuilder);
 	}
 
 	create(parent: HTMLElement): void {
 		addClass(parent, 'notebookExplorer-viewlet');
 		this.root = parent;
 
+		this.searchWidgetsContainerElement = append(this.root, $('.header'));
+		this.createSearchWidget(this.searchWidgetsContainerElement);
+
 		this.notebookSourcesBox = append(this.root, $('.notebookSources'));
 
 		return super.create(this.notebookSourcesBox);
 	}
 
+	private createSearchWidget(container: HTMLElement): void {
+		this.searchWidget = this._register(this.instantiationService.createInstance(NotebookSearchWidget, container, <INotebookExplorerSearchOptions>{
+			value: '',
+			replaceValue: undefined,
+			isRegex: false,
+			isCaseSensitive: false,
+			isWholeWords: false,
+			searchHistory: [],
+			replaceHistory: [],
+			preserveCase: false
+		}));
+
+
+		this._register(this.searchWidget.onSearchSubmit(options => this.triggerQueryChange(options)));
+		this._register(this.searchWidget.onSearchCancel(({ focus }) => this.cancelSearch(focus)));
+		this._register(this.searchWidget.searchInput.onDidOptionChange(() => this.triggerQueryChange()));
+
+		this._register(this.searchWidget.onDidHeightChange(() => this.searchView?.reLayout()));
+
+		this._register(this.searchWidget.onPreserveCaseChange((state) => {
+			if (this.searchView && this.searchView.searchViewModel) {
+				this.searchView.searchViewModel.preserveCase = state;
+			}
+			this.refreshTree();
+		}));
+
+		this._register(this.searchWidget.searchInput.onInput(() => this.searchView?.updateActions()));
+
+		this.trackInputBox(this.searchWidget.searchInputFocusTracker);
+	}
+
+	cancelSearch(focus: boolean = true): boolean {
+		if (focus) {
+			this.searchWidget.focus();
+			return true;
+		}
+		return false;
+	}
+
+	triggerQueryChange(_options?: { preserveFocus?: boolean, triggeredOnType?: boolean, delay?: number }) {
+		const options = { preserveFocus: true, triggeredOnType: false, delay: 0, ..._options };
+
+		if (!this.pauseSearching) {
+			this.triggerQueryDelayer.trigger(() => {
+				this._onQueryChanged(options.preserveFocus, options.triggeredOnType);
+			}, options.delay);
```

Comment:
Does this ever get cleaned up? Is there a chance we can get into a state where this is still being called without it being present on the UI?

Reply:
am initializing this by calling this._register which is a Disposable register. I don't think we can get into that state.